tests/scripts/expect/_common.exp

```diff
@@ -168,7 +170,7 @@ proc get_extaddr {} {
 proc get_rloc16 {} {
     send "rloc16\n"
     expect "rloc16"
-    expect -re {([0-9a-fA-F]{4})}
+    expect -re {[\r\n ]([0-9a-fA-F]{4})[\r\n>]}
     set rval $expect_out(1,string)
     expect "Done"
```

Comment:
```suggestion
    expect -re "{[\r\n ]Done[\r\n]}"
```
is it necessary to do so?

Reply:
I realize that this is problematic because the  `\r\n` between extended address output and Done would have been consumed by `expect -re {[\r\n ]([0-9a-fA-F]{4})[\r\n>]}`, so `"{[\r\n ]Done[\r\n]}"` won't match the next line of `Done`.

I am fine to keep it as is.

Reply:
We can use a positive lookahead to assert the existence of newline without consuming it.

E.g.
```
expect -re {[\r\n ]([0-9a-fA-F]{4})(?=[\r\n>])}
```

`expect` doesn't seem to support lookbehind though. So only the newline after the pattern can be asserted this way, the newline before the pattern has to be comsumed.

Reply:
OpenThread prints `\r\n` for line break, `[\r\n>]` consumes at most a single `\r`. And lookbehind looks like a safer solution.